HyperVExtension/test/HyperVExtension/HyperVExtensionTests/Services/HyperVExtensionIntegrationTest.cs

```diff
@@ -361,4 +378,118 @@ public void TestDevSetupAgentDeployment()
+        var inputJson = JsonSerializer.Serialize(new VMGalleryCreationUserInput()
+        {
+            NewVirtualMachineName = expectedVMName,
+
+            // Get Image with the smallest size from gallery, we'll use it to create a VM.
+            SelectedImageListIndex = smallestImageIndex,
+        });
+
+        var createComputeSystemOperation = hyperVProvider.CreateCreateComputeSystemOperation(null, inputJson);
+        createComputeSystemOperation!.Progress += OnProgressReceived;
+
+        // Act
+        var createComputeSystemResult = await createComputeSystemOperation!.StartAsync();
+        createComputeSystemOperation!.Progress -= OnProgressReceived;
+
+        // Assert
+        Assert.AreEqual(ProviderOperationStatus.Success, createComputeSystemResult.Result.Status);
+        Assert.AreEqual(expectedVMName, createComputeSystemResult.ComputeSystem.DisplayName);
+        CleanUpVirtualMachine(createComputeSystemResult.ComputeSystem, imageList.Images[smallestImageIndex], expectedVMName);
+    }
+
+    public void CleanUpVirtualMachine(IComputeSystem computeSystem, VMGalleryImage image, string virtualMachineName)
+    {
+        try
+        {
+            var hyperVManager = TestHost!.GetService<IHyperVManager>();
+            var virtualMachine = computeSystem as HyperVVirtualMachine;
+            var hardDisk = virtualMachine!.GetHardDrives().First();
+
+            // remove extracted VHD
+            File.Delete(hardDisk!.Path!);
+            var temp = Path.GetTempPath();
+
+            // remove extracted archive
+            var archiveFileExtension = Path.GetExtension(image.Disk.ArchiveRelativePath);
+            var fileName = image.Disk.Hash.Split(":").Last();
+            var archiveFilePath = Path.Combine(temp, $"{fileName}.{archiveFileExtension}");
+            File.Delete(archiveFilePath);
+
+            // remove virtual machine
+            hyperVManager.RemoveVirtualMachine(Guid.Parse(computeSystem.Id));
+        }
+        catch (Exception ex)
+        {
+            Debug.WriteLine($"Failed to clean up virtual machine with name {virtualMachineName}: {ex}");
+        }
+    }
+
+    public async Task<int> GetIndexOfImageWithSmallestRequiredSpace(VMGalleryImageList imageList)
+    {
+        var smallestImageIndex = 0;
+        var smallestSize = ulong.MaxValue;
+        var httpClient = TestHost!.GetService<IHttpClientFactory>()!.CreateClient();
+        for (var i = 0; i < imageList.Images.Count; i++)
+        {
+            var sourceWebUri = new Uri(imageList.Images[i].Disk.Uri);
+            var response = await httpClient.GetAsync(sourceWebUri, HttpCompletionOption.ResponseHeadersRead);
+            var totalSizeOfDisk = response.Content.Headers.ContentLength ?? 0L;
+            if (ulong.TryParse(imageList.Images[i].Requirements.DiskSpace, CultureInfo.InvariantCulture, out var requiredDiskSpace))
+            {
+                if ((requiredDiskSpace + (ulong)totalSizeOfDisk) < smallestSize)

```

Comment:
How are totalSizeOfDisk and requiredDiskSpace related to each other? I'd think that requiredDiskSpace includes totalSizeOfDisk. If not, could you add a comment about that.

Reply:
Interestingly the Hyper-V folks are using both to show the download size in the quick create window when I looked at the code. I agree its a bit unintuitive but I added a comment about it. Thanks.